resource/user/user.go

```diff
@@ -401,6 +336,40 @@ func (u *User) DiffAdd(status *resource.Status) (*AddUserOptions, error) {
 	return options, nil
 }
 
+// DiffDel checks for differences between the current and desired state for the
+// user to be deleted indicated by the User fields.
+func (u *User) DiffDel(status *resource.Status, userByName *user.User, nameNotFound bool) error {
```

Comment:
IMO it would be better to make the tests part of the package than it would be to expose extra members. But it's definitely not a blocker. :)